lib/collection/src/shards/transfer/stream_records.rs

```diff
@@ -62,12 +63,33 @@ pub(super) async fn transfer_stream_records(
             .transfer_batch(offset, TRANSFER_BATCH_SIZE)
             .await?;
 
+        // If this is the last batch, finalize
         if offset.is_none() {
-            // That was the last batch, all look good
             break;
         }
     }
 
+    // Update cutoff point on remote shard, disallow recovery before our current last seen
+    // Allow but warn about errors, the remote shard may not support this operation
+    {
+        let shard_holder = shard_holder.read().await;
+        let Some(replica_set) = shard_holder.get_shard(&shard_id) else {
+            // Forward proxy gone?!
+            // That would be a programming error.
+            return Err(CollectionError::service_error(format!(
+                "Shard {shard_id} is not found"
+            )));
+        };
+
+        let cutoff = replica_set.shard_recovery_point().await?;
+        let result = remote_shard
+            .update_shard_cutoff_point(collection_name, shard_id, &cutoff)
+            .await;
+        if let Err(err) = result {
+            log::warn!("Failed to update cutoff point on remote shard, ignoring: {err}");
+        }
```

Comment:
Should we explicitly only allow "not implemented" errors here? 🤔

Reply:
There should be specific `tonic::Status` or something like this. IIRC, I've done it for `/readyz`.

Reply:
There is, but I'm afraid it is abstracted away here.

Reply:
Should we add a `TODO: Only ignore "API unimplemented" error!` here or something?

Reply:
Oh, before merging, please this! 🙏

Reply:
I have implemented it like this now:

```rust
        // Warn and ignore if remote shard is running an older version, error otherwise
        match result {
            Err(err) if err.to_string().starts_with("Service internal error: Tonic status error: status: Unimplemented") => {
                log::warn!("Cannot update cutoff point on remote shard because it is running an older version, ignoring: {err}");
            }
            Err(err) => return Err(err),
            Ok(()) => {},
        }
```

I don't like it at all, and it may be fragile, but I don't see a better way.

Reply:
Oh, sorry, I just meant to add a `TODO`... Sorry! 🤦‍♀️

I think we can fix this in a more meaningful way later. Like, add another error variant or something.

Reply:
I'll put the TODO back to make sure we don't miss it.